terraform/services/snyk-integration/main.tf

```diff
@@ -40,28 +47,34 @@ data "aws_iam_policy_document" "snyk_pull" {
       "ecr:ListTagsForResource",
       "ecr:ListImages",
       "ecr:BatchCheckLayerAvailability",
-      "ecr:GetRepositoryPolicy",
-      "ecr:GetLifecyclePolicy"
+      "ecr:GetRepositoryPolicy"
+    ]
+    resources = [
+      "arn:aws:ecr:us-east-1:${data.aws_caller_identity.current.account_id}:repository/${each.key}-*"
     ]
-    resources = ["*"]
   }
 }
 
 resource "aws_iam_policy" "snyk_pull" {
```

Comment:
Switch this to `aws_iam_role_policy` to create an inline policy. We don't need managed policies for each of these roles.